sdk/core/azure_core_amqp/src/fe2o3/cbs.rs

```diff
@@ -115,13 +112,15 @@ impl AmqpClaimsBasedSecurityApis for Fe2o3ClaimsBasedSecurity<'_> {
             .borrow_mut()
             .put_token(path, cbs_token)
             .await
-            .map_err(|e| {
-                let me = azure_core::Error::try_from(Fe2o3ManagementError(e));
-                if let Err(e) = me {
+            .map_err(|e| match AmqpError::try_from(e) {
+                Ok(amqp_error) => amqp_error.into(),
+                Err(e) => {
                     debug!("Failed to convert management error to azure error: {:?}", e);
-                    return e;
+                    azure_core::Error::message(
+                        azure_core::error::ErrorKind::Amqp,
+                        format!("Failed to convert management error to azure error: {}", e),
+                    )
```

Comment:
I'd use `Error::full` to wrap the error either in addition to formatting it or in lieu of. We're kinda inconsistent throughout the SDK now with which.

Reply:
Fair point. I'll fix it in the next iteration.